From 26182560fd837130dde97b3107418c5e320937e2 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Sun, 30 Sep 2018 18:32:55 -0700
Subject: [PATCH 4/7] DO-NOT-UPSTREAM: Disable some configs with Clang

* CPU_BIG_ENDIAN: Not concerned with big endian at this point and it
  causes some additional warnings.
  Links: https://github.com/ClangBuiltLinux/linux/issues/36
         https://github.com/ClangBuiltLinux/linux/issues/214

* GCOV_KERNEL: Patches currently in testing, just don't want to add
  them to my stack.
  Link: https://github.com/ClangBuiltLinux/linux/issues/17

* HAVE_FUNCTION_TRACER on arm: Broken with Clang.
  Link: https://github.com/ClangBuiltLinux/linux/issues/35

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 arch/arm/Kconfig    | 2 +-
 arch/arm/mm/Kconfig | 2 +-
 arch/arm64/Kconfig  | 1 +
 kernel/gcov/Kconfig | 2 +-
 4 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 664e918e2624..73e970f2bf75 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -70,7 +70,7 @@ config ARM
 	select HAVE_EXIT_THREAD
 	select HAVE_FTRACE_MCOUNT_RECORD if !XIP_KERNEL
 	select HAVE_FUNCTION_GRAPH_TRACER if !THUMB2_KERNEL
-	select HAVE_FUNCTION_TRACER if !XIP_KERNEL
+	select HAVE_FUNCTION_TRACER if !XIP_KERNEL && !CC_IS_CLANG
 	select HAVE_GCC_PLUGINS
 	select HAVE_GENERIC_DMA_COHERENT
 	select HAVE_HW_BREAKPOINT if PERF_EVENTS && (CPU_V6 || CPU_V6K || CPU_V7)
diff --git a/arch/arm/mm/Kconfig b/arch/arm/mm/Kconfig
index b169e580bf82..63fdc70552aa 100644
--- a/arch/arm/mm/Kconfig
+++ b/arch/arm/mm/Kconfig
@@ -741,7 +741,7 @@ config SWP_EMULATE
 
 config CPU_BIG_ENDIAN
 	bool "Build big-endian kernel"
-	depends on ARCH_SUPPORTS_BIG_ENDIAN
+	depends on ARCH_SUPPORTS_BIG_ENDIAN && !CC_IS_CLANG
 	help
 	  Say Y if you plan on running a kernel in big-endian mode.
 	  Note that your board must be properly built and your board
diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index cfbf307d6dc4..9c0213ff45c7 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -772,6 +772,7 @@ config ARM64_PA_BITS
 
 config CPU_BIG_ENDIAN
        bool "Build big-endian kernel"
+       depends on !CC_IS_CLANG
        help
          Say Y if you plan on running a kernel in big-endian mode.
 
diff --git a/kernel/gcov/Kconfig b/kernel/gcov/Kconfig
index 1e3823fa799b..8d3dc2303618 100644
--- a/kernel/gcov/Kconfig
+++ b/kernel/gcov/Kconfig
@@ -2,7 +2,7 @@ menu "GCOV-based kernel profiling"
 
 config GCOV_KERNEL
 	bool "Enable gcov-based kernel profiling"
-	depends on DEBUG_FS
+	depends on DEBUG_FS && !CC_IS_CLANG
 	select CONSTRUCTORS if !UML
 	default n
 	---help---
-- 
2.20.1

